### PR TITLE
Wait for elasticsearch cluster to be up

### LINKED ISSUE #216: Wait for ES deployment to be ready and then deploy jaeger
Now operator creates ES CR and jaeger objects (deployments) at the same time. This is causing jaeger pods to fail multiple times until Elasticsearch is up.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #216 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/242)

**Comment by pavolloffay** [comment]
There is one minor is `deployment.go` deletes ES deployment once it's up, not sure why I will have to dig into it

**Comment by pavolloffay** [comment]
> Do you mean that our Operator removes the Elasticsearch stateful sets?

There are no SS but yes. These objects would be queried by our controllers - which should not be bc they are managed by ES operator

File: pkg/controller/jaeger/elasticsearch.go [hunk 1]
```diff
@@ -2,7 +2,12 @@ package jaeger
 
 import (
 	"context"
+	"time"
 
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
```
Comment (jpkrohling) [inline pkg/controller/jaeger/elasticsearch.go]: Across the code base, we are now using `log` as alias to this package.

File: pkg/controller/jaeger/elasticsearch.go [hunk 2]
```diff
@@ -44,3 +52,30 @@ func (r *ReconcileJaeger) applyElasticsearches(jaeger v1alpha1.Jaeger, desired [
...
+		if err = c.List(context.Background(), client.MatchingLabels(es.Labels).InNamespace(es.Namespace), &depList); err != nil {
+			return false, err
+		}
+		available := int32(0)
+		for _, d := range depList.Items {
+			if d.Status.Replicas == d.Status.AvailableReplicas {
+				available++
+			}
+		}
+		logrus.WithFields(logrus.Fields{
```
Comment (jpkrohling) [inline pkg/controller/jaeger/elasticsearch.go]: `s/logrus/log`

**Pull request merged**